src/fixes.rs

```diff
@@ -346,3 +386,113 @@ fn find_use_path_list(db: &dyn SyntaxGroup, node: &SyntaxNode) -> SyntaxNode {
         .find(|descendant| descendant.kind(db) == SyntaxKind::UsePathList)
         .unwrap_or(*node)
 }
+
+pub fn merge_overlapping_fixes(
+    db: &mut FixerDatabase,
+    file_id: FileId,
+    fixes: Vec<Fix>,
+) -> Vec<Fix> {
+    let mut current_fixes: Vec<Fix> = fixes.clone();
+    let mut were_overlapped = false;
+    let file_content = db.file_content(file_id).unwrap();
+
+    while let Some(overlapping_fix) = get_first_overlapping_fix(&current_fixes) {
+        were_overlapped = true;
+        apply_fix_for_file(db, file_id, overlapping_fix);
+
+        let diags: Vec<SemanticDiagnostic> = db
+            .file_modules(file_id)
+            .unwrap()
+            .iter()
+            .filter_map(|module_id| db.module_semantic_diagnostics(*module_id).ok())
+            .flat_map(|diag| diag.get_all())
+            .filter(|diag| diag.stable_location.diagnostic_location(db).file_id == file_id)
+            .collect();
+
+        current_fixes = get_fixes_without_resolving_overlapping(db, diags)
+            .values()
+            .flat_map(|v| v.clone())
+            .collect();
+    }
+
+    if were_overlapped {
+        for fix in current_fixes.iter() {
+            apply_fix_for_file(db, file_id, fix.clone());
+        }
+        let file_content_after = db.file_content(file_id).unwrap();
+        // Currently we are just replacing the entire file content with the new fixed one.
+        // This is not ideal, but as for now we don't need to worry about it.
+        current_fixes = vec![Fix {
+            span: TextSpan {
+                start: TextOffset::START,
+                end: TextWidth::from_str(&file_content).as_offset(),
+            },
+            suggestion: file_content_after.to_string(),
+        }];
```

Comment:
When we decide upon it being correct for now, i will create a separate issue to address making granular mapped fixes to old file